packages/Search/examples/viz/tree_visualization.cpp

```diff
@@ -0,0 +1,157 @@
+ ****************************************************************************/
+
+#include <DTK_DetailsTreeVisualization.hpp>
+#include <DTK_LinearBVH.hpp>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_DefaultNode.hpp>
+
+#include <boost/program_options.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <random>
+
+#include <point_clouds.hpp>
+
+template <typename View>
+void printPointCloud( View points, std::ostream &os )
+{
+    auto const n = points.extent_int( 0 );
+    for ( int i = 0; i < n; ++i )
+        os << "\\node[leaf] at (" << points( i )[0] << "," << points( i )[1]
+           << ") {\\textbullet};\n";
+}
+
+template <typename TreeType>
+void viz( std::string const &prefix, std ::string const &infile,
+          int n_neighbors )
+{
+    using DeviceType = typename TreeType::device_type;
+    using ExecutionSpace = typename DeviceType::execution_space;
+    Kokkos::View<DataTransferKit::Point *, DeviceType> points( "points" );
+    loadPointCloud( infile, points );
+
+    TreeType bvh( points );
+
+    using TreeVisualization =
+        typename DataTransferKit::Details::TreeVisualization<DeviceType>;
+    using TikZVisitor = typename TreeVisualization::TikZVisitor;
+    using GraphvizVisitor = typename TreeVisualization::GraphvizVisitor;
+
+    int const n_queries = bvh.size();
+    if ( n_neighbors < 0 )
+        n_neighbors = bvh.size();
+    Kokkos::View<DataTransferKit::Nearest<DataTransferKit::Point> *, DeviceType>
+        queries( "queries", n_queries );
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
+                          KOKKOS_LAMBDA( int i ) {
+                              queries( i ) = DataTransferKit::nearest(
+                                  points( i ), n_neighbors );
+                          } );
+    Kokkos::fence();
+
+    auto performQueries = [&bvh, &queries]( std::string const &p,
+                                            std::string const &s ) {
+        std::ofstream fout;
+        for ( int i = 0; i < queries.extent_int( 0 ); ++i )
+        {
+            std::string const fname = p + std::to_string( i ) + s;
+            fout.open( fname, std::fstream::out );
+            TreeVisualization::visit( bvh, queries( i ),
+                                      GraphvizVisitor{fout} );
+            fout.close();
+        }
+    };
+
+    std::fstream fout;
+
+    // Print the point cloud
+    fout.open( prefix + "points.tex", std::fstream::out );
+    printPointCloud( points, fout );
+    fout.close();
+
+    // Print the bounding volume hierarchy
+    fout.open( prefix + "bounding_volumes.tex", std::fstream::out );
+    TreeVisualization::visitAllIterative( bvh, TikZVisitor{fout} );
+    fout.close();
+
+    // Print the entire tree
+    fout.open( prefix + "tree_all_nodes_and_edges.dot.m4", std::fstream::out );
+    TreeVisualization::visitAllIterative( bvh, GraphvizVisitor{fout} );
```

Comment:
Are the entire tree and the bvh in different format?

Reply:
You mean the file produced by the `TikZGraphvisitor`?

Reply:
yes

Reply:
`GraphvizVisitor` produces nodes and edges statements that would be listed for a graph in DOT format.
`TikZGraphvisitor` produces command you would enclose in a `tikzpicture` in a LateX document.

Reply:
I added comments to capture this.